circle.yml

```diff
@@ -0,0 +1,208 @@
+version: 2
+
+workflows:
+  version: 2
+  default:
+    jobs:
+      - prepare:
+          filters:
+            tags:
+              only: /.*/
+      - lint:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - build:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-flow:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-unit:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-render:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-query:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-expressions:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - deploy-benchmarks:
+          requires:
+            - lint
+            - build
+            - test-flow
+            - test-unit
+            - test-render
+            - test-query
+            - test-expressions
+          filters:
+            tags:
+              only: /v[0-9]+.[0-9]+.[0-9]+(-.+)?/
+            branches:
+              only:
+                - master
+                - /release-.*/
+      - deploy-release:
+          requires:
+            - lint
+            - build
+            - test-flow
+            - test-unit
+            - test-render
+            - test-query
+            - test-expressions
+          filters:
+            tags:
+              only: /v[0-9]+.[0-9]+.[0-9]+(-.+)?/
+            branches:
+              ignore: /.*/
+
+defaults: &defaults
+  docker:
+    - image: mbgl/ci:r4-linux-gl-js
+  working_directory: ~/mapbox-gl-js
+
+jobs:
+  prepare:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-yarn-{{ checksum "yarn.lock" }}
+            - v1-yarn
+      - run: yarn
+      - save_cache:
+          key: v1-yarn-{{ checksum "yarn.lock" }}
+          paths:
+            - '~/.yarn'
+            - 'node_modules'
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+
+  lint:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: .
+      - restore_cache:
+          keys:
+            - v1-lint-{{ .Branch }}
+            - v1-lint
+      - run: yarn run lint
+      - run: yarn run lint-docs
+      - run: yarn run lint-css
+      - save_cache:
+          key: v1-lint-{{ .Branch }}-{{ .Revision }}
```

Comment:
Looks like we never restore a cache with this key.  Should this match the `restore_cache` key above?

Reply:
No; restoration uses prefix matching, with the most recent winning in case of ties, and caches are immutable once written.

Reply:
Oh, that's nice!